include/materials/modified_cam_clay.tcc

```diff
@@ -267,51 +269,13 @@ template <unsigned Tdim>
 bool mpm::ModifiedCamClay<Tdim>::compute_stress_invariants(
     const Vector6d& stress, mpm::dense_map* state_vars) {
   // Compute volumetic stress
-  (*state_vars).at("p") = -(stress(0) + stress(1) + stress(2)) / 3.;
-  // Compute the deviatoric stress
-  Vector6d dev_stress = Vector6d::Zero();
-  dev_stress(0) = stress(0) + (*state_vars).at("p");
-  dev_stress(1) = stress(1) + (*state_vars).at("p");
-  dev_stress(2) = stress(2) + (*state_vars).at("p");
-  dev_stress(3) = stress(3);
-  if (Tdim == 3) {
-    dev_stress(4) = stress(4);
-    dev_stress(5) = stress(5);
-  }
-  // Compute J2
-  double j2 = (std::pow((stress(0) - stress(1)), 2) +
-               std::pow((stress(1) - stress(2)), 2) +
-               std::pow((stress(0) - stress(2)), 2)) /
-                  6.0 +
-              std::pow(stress(3), 2);
-  if (Tdim == 3) j2 += std::pow(stress(4), 2) + std::pow(stress(5), 2);
-  // Compute q
-  (*state_vars).at("q") = std::sqrt(3. * j2);
-
-  if (three_invariants_) {
-    // Compute J3
-    (*state_vars).at("j3") = (dev_stress(0) * dev_stress(1) * dev_stress(2)) -
-                             (dev_stress(2) * std::pow(dev_stress(3), 2));
-    if (Tdim == 3)
-      (*state_vars).at("j3") +=
-          ((2 * dev_stress(3) * dev_stress(4) * dev_stress(5)) -
-           (dev_stress(0) * std::pow(dev_stress(4), 2)) -
-           (dev_stress(1) * std::pow(dev_stress(5), 2)));
-    // Compute theta value (Lode angle)
-    double theta_val = 0.;
-    if (std::fabs(j2) > 0.0)
-      theta_val = (3. * std::sqrt(3.) / 2.) *
-                  ((*state_vars).at("j3") / std::pow(j2, 1.5));
-    // Check theta value
-    if (theta_val > 1.0) theta_val = 1.0;
-    if (theta_val < -1.0) theta_val = -1.0;
-    // Compute theta
-    (*state_vars).at("theta") = (1. / 3.) * acos(theta_val);
-    // Check theta
-    if ((*state_vars).at("theta") > M_PI / 3.)
-      (*state_vars).at("theta") = M_PI / 3.;
-    if ((*state_vars).at("theta") < 0.0) (*state_vars).at("theta") = 0.;
-  }
+  (*state_vars).at("p") = -mpm::materials::p(stress);
+  // Compute deviatoric q
+  (*state_vars).at("q") = mpm::materials::q(stress);
+  // Compute theta (Lode angle)
+  if (three_invariants_)
+    (*state_vars).at("theta") = mpm::materials::lode_angle(
+        stress, std::numeric_limits<double>::epsilon());
```

Comment:
Suggestion to add optional `tolerance` value users can specify.
Other than that, you can use `lode_angle` with default tolerance value.

Reply:
Well I think there is no rule to choose it, so I make it implicit. But if you want it to be an option @kks32 @ezrayst , I can do it.

Reply:
No need actually if you feel this model does not need. If so, you don't have to put tolerance argument to use this function. Please see my suggestion change of this line.

Reply:
I see, agree with it.